File: leetcode/maychallenge/isPerfectSquare.go
package main

// func main() {
// 	fmt.Println(isPerfectSquare(16))
// }

func isPerfectSquare(num int) bool {
	for i := 1; i <= num; i++ {
		if i*i > num {
			return false
		}
		if float64(num)/float64(i) == float64(i) {
			return true
		}
	}
	return false
}

File: leetcode/MayLeetCodePy/isPerfectSquare.py
# isPerfectSquare
def isPerfectSquare(num: int) -> bool:
    # for i in range(num+1):
    #      if i > 0:  
    #         if i*i > num:
    #             return False
    #         if num/i == i:
    #             return True
    # return False
    a = num**0.5
    print(a)
    print(int(a))
    if a == int(a):
        return True
    else:
        return False

print(isPerfectSquare(4))